src/pipelines/question_answering.rs

```diff
@@ -914,23 +916,18 @@ impl QuestionAnsweringModel {
             .iter()
             .map(|feature| &feature.input_ids)
             .map(|input| {
-                let mut attention_mask = vec![1; input.len()];
-                attention_mask.append(&mut vec![0; max_len - attention_mask.len()]);
+                let mut attention_mask = Vec::with_capacity(max_len);
+                attention_mask.resize(input.len(), 1);
+                attention_mask.resize(max_len, 0);
                 attention_mask
             })
             .map(|input| Tensor::of_slice(&(input)))
             .collect::<Vec<_>>();
 
         for feature in features.iter_mut() {
-            feature
-                .offsets
-                .append(&mut vec![None; max_len - feature.input_ids.len()]);
```

Comment:
This is padding `feature.offsets` but is referencing `feature.input_ids`. I imagine the idea is that `feature.offsets.len() == feature.input_ids.len()` and we want everything to be `max_len` long. If that's not correct then whoops!

Reply:
This is correct